base/lnls-prune-artifacts.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+set -eu
+
+target_paths=$@
+
+
+get_used_epics_modules() {
+    targets=$@
+
+    executables=$(find $targets -type f -executable -print)
+    libs="$(ldd $executables 2>/dev/null)"
+
+    modules=()
+    epics_libs=$(echo "$libs" | grep -oE "$EPICS_MODULES_PATH/.*/*.so")
+    for lib in $epics_libs; do
+        module=${lib%/lib/*/*lib*so}
+        echo $module
+    done | sort | uniq
+}
+
+get_all_epics_modules() {
+    release_defs=$(cat ${EPICS_RELEASE_FILE} | grep = | cut -d'=' -f 2)
+
+    modules="$(echo "$release_defs" | grep $EPICS_MODULES_PATH)"
+
+    echo "$modules"
+}
+
+remove_unused_epics_modules() {
+    targets=$@
+
+    used_modules=$(get_used_epics_modules $targets)
+
+    for module in $(get_all_epics_modules); do
+        if [[ -d $module && ! $used_modules =~ "$module" ]]; then
+            size=$(du -hs $module | cut -f 1)
```

Comment:
Why is `-d $module` required? Shouldn't they always exist? Or is this covering some case like an AreaDetector or motor submodule being in the list after the super-repository has already been cleaned? I think this should be explained somewhere.

I find `=~` somewhat non obvious to parse. And it means "motor" won't be removed if "motorX" is in used modules, right? I think it might make sense to print the list into a file (in /tmp, which can be removed afterwards), so it's split by newlines, and use grep to match the *whole* line.

Reply:
> And it means "motor" won't be removed if "motorX" is in used modules

Because of such issues with prefix matching, I moved to checking the each level of the path at a time, with whole line matching instead. This turned out to be handy for the following patches as well.

Still, don't feel like my implementation explores all bash-ims it could.